src/main/java/net/milkbowl/vault2/economy/Economy.java

```diff
@@ -0,0 +1,324 @@
+     */
+    public boolean isEnabled();
+
+    /**
+     * Gets name of economy plugin.
+     * 
+     * @return Name of Economy plugin.
+     */
+    public String getName();
+
+    /**
+     * Returns true if the given implementation supports banks.
+     * 
+     * @return true if the implementation supports banks
+     */
+    public boolean hasBankSupport();
+
+    /*
+     * Currency-related methods follow.
+     */
+
+    /**
+     * Some economy plugins round off after a certain number of digits. This
+     * function returns the number of digits the plugin keeps or -1 if no rounding
+     * occurs.
+     * 
+     * @return number of digits after the decimal point kept
+     */
+    public int fractionalDigits();
+
+    /**
+     * Format amount into a human readable String This provides translation into
+     * economy specific formatting to improve consistency between plugins.
+     *
+     * @param amount to format
+     * @return Human readable string describing amount
+     */
+    public String format(double amount);
+
+    /**
+     * Returns the name of the currency in plural form. If the economy being used
+     * does not support currency names then an empty string will be returned.
+     * 
+     * @return name of the currency (plural)
+     */
+    public String currencyNamePlural();
+
+    /**
+     * Returns the name of the currency in singular form. If the economy being used
+     * does not support currency names then an empty string will be returned.
+     * 
+     * @return name of the currency (singular)
+     */
+    public String currencyNameSingular();
+
+    /*
+     * Account-related methods follow.
+     */
+
+    /**
+     * Attempts to create a account for the given uuid
+     * 
+     * @param uuid associated with the account
+     * @param name associated with the account.
+     * @return if the account creation was successful
+     */
+    public boolean createAccount(UUID uuid, String name);
+
+    /**
+     * Attempts to create an account for the given UUID on the specified world
+     * IMPLEMENTATION SPECIFIC - if an economy plugin does not support this then
+     * false will always be returned.
+     * 
+     * @param uuid      associated with the account
+     * @param name      associated with the account.
+     * @param worldName String name of the world
+     * @return if the account creation was successful
+     */
+    public boolean createAccount(UUID uuid, String name, String worldName);
+
+    /**
+     * Returns a map that represents all of the UUIDs which have accounts in the
+     * plugin, as well as their last-known-name whether it is null or not. This is
+     * used for Vault's economy converter and should be given every account
+     * available.
+     * 
+     * @return a {@link Map} composed of the accounts keyed by their UUID, along
+     *         with their associated last-known-name (null or not.)
+     */
+    public Map<UUID, @Nullable String> getUUIDNameMap();
```

Comment:
This shouldn't be possible to be null correct? The createAccount methods require a name and it could be arbitrary too. So I could create an account with even a random UUID (not from a player) but I'd still have provide a name, such as the "Cereal Shop". I think we can drop the @Nullable annotation here.

Along with that, should be define a method to rename an account?

Reply:
Removed the annotation.

Reply:
Added a renameAccount(UUID, String) method.